File: Scala/src/main/scala/org/revian/MyStack.scala
package org.revian

class MyStack {
  private var stack = List[Int]()

  def push(value: Int): Unit = {
    stack = value :: stack
  }

  def pop(): Int = {
    if (stack.isEmpty) throw new IllegalStateException("Stack is empty")
    val top = stack.head
    stack = stack.tail
    top
  }

  def size(): Int = stack.length
}

File: Java/src/main/java/org/revian/MyStack.java
package org.revian;

public class MyStack {
    private int[] stack;
    private int size;
    private static final int DEFAULT_CAPACITY = 10;

    public MyStack() {
        stack = new int[DEFAULT_CAPACITY];
        size = 0;
    }

    public void push(int value) {
        if (size >= stack.length) {
            throw new StackOverflowError("Stack is full");
        }
        stack[size++] = value;
    }

    public int pop() {
        if (size == 0) {
            throw new IllegalStateException("Stack is empty");
        }
        return stack[--size];
    }

    public int peek() {
        if (size == 0) {
            throw new IllegalStateException("Stack is empty");
        }
        return stack[size - 1];
    }

    public int size() {
        return size;
    }
}

